src/Agent.Worker/StepsRunner.cs

```diff
@@ -107,6 +111,12 @@ public async Task RunAsync(IExecutionContext jobContext, IList<IStep> steps)
                             ConditionResult conditionReTestResult;
                             if (HostContext.AgentShutdownToken.IsCancellationRequested)
                             {
+                                if (AgentKnobs.FailJobWhenAgentDies.GetValue(jobContext).AsBoolean())
+                                {
+                                    jobContext.Result = TaskResult.Failed;
+                                    jobContext.Variables.Agent_JobStatus = jobContext.Result;
+                                }
+                                PublishTelemetry (jobContext, jobContext.Result.ToString(), "120");
```

Comment:
>PublishTelemetry [](http://example.com/codeflow?start=32&length=16)

Please make this consistent; in other locations, telemetry is inside the knob.

Reply:
Done